ejmask-extensions/src/test/java/com/ebay/ejmask/extenstion/builder/json/JsonNumericFieldPatternBuilderTest.java

```diff
@@ -0,0 +1,185 @@
+package com.ebay.ejmask.extenstion.builder.json;
+
+import com.ebay.ejmask.api.PatternEntity;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+/**
+ * @author fsun1
+ */
+public class JsonNumericFieldPatternBuilderTest {
+
+    public static final JsonNumericFieldPatternBuilder instance = new JsonNumericFieldPatternBuilder();
+    public static final String[] fieldNames = new String[]{"firstName", "lastName", "number", "boolValue"};
+
+
+    /**
+     * Test of buildPattern method, of class JsonFieldPatternBuilder.
+     */
+    @Test
+    public void testBuildPattern() {
+        int visibleCharacters = 12;
+        String result = instance.buildPattern(visibleCharacters, fieldNames);
+        Assertions.assertEquals("\\\"(firstName|lastName|number|boolValue)(\\\\*\\\"\\s*:\\s*\\\\*)(-?\\b\\d+(\\.\\d+)?(e-?\\d+)?\\b)([^\\\"]{1,2})", result);
+    }
+
+    /**
+     * Test of buildPattern method, of class JsonFieldPatternBuilder.
+     */
+    @Test
+    public void testBuildPattern_without_visible_characters() {
+        int visibleCharacters = 0;
+        Assertions.assertThrows(IllegalArgumentException.class, () -> instance.buildPattern(visibleCharacters, fieldNames));
+    }
+
+    /**
+     * Test of buildReplacement method, of class JsonFieldPatternBuilder.
+     */
+    @Test
+    public void testBuildReplacement() {
+        int visibleCharacters = 0;
+        String result = instance.buildReplacement(visibleCharacters, fieldNames);
+        Assertions.assertEquals("\"$1$2\"xxxx\"$6", result);
+    }
+
+    @ParameterizedTest
+    @MethodSource("dataForTestMatch")
+    public void testMatch(String name, String data, String expected) {
+        String regex = instance.buildPattern(2, fieldNames);
+        String replacement = instance.buildReplacement(2, fieldNames);
+        Pattern pattern = Pattern.compile(regex);
+        String result = pattern.matcher(data).replaceAll(replacement);
+        Assertions.assertEquals(expected, result);
+    }
+
+    @ParameterizedTest
+    @MethodSource("dataForTestMatch")
+    public void testMatchForPatternList(String name, String data, String expected) {
```

Comment:
what is the usage of name at here?  do you want to put it in the test case name of `@ParameterizedTest`?

```java
@ParameterizedTest(name = "{0}")
```

Reply:
it is not used by asserting itself, but for issue triage, the console would show the name in case a test is failed